fs/btrfs/compression.h

```diff
@@ -119,7 +120,9 @@ struct btrfs_compress_op {
 			  unsigned long start_byte,
 			  size_t srclen, size_t destlen);
 
-	void (*set_level)(struct list_head *ws, unsigned int type);
+	int (*set_level)(struct list_head *ws, unsigned level);
```

Comment:
You should add a comment that describes what this function is, and what the return value means.